zebra-network/src/meta_addr.rs

```diff
@@ -79,6 +79,39 @@ impl PeerAddrState {
             AttemptPending | Responded | Failed => false,
         }
     }
+
+    /// Returns the typical connection state machine order of `self` and `other`.
+    /// Partially ordered states are sorted in connection attempt order.
+    ///
+    /// See [`MetaAddrChange::apply_to_meta_addr()`] for more details.
+    fn connection_state_order(&self, other: &Self) -> Ordering {
+        use Ordering::*;
+        match (self, other) {
+            (NeverAttemptedAlternate, NeverAttemptedAlternate)
+            | (NeverAttemptedGossiped, NeverAttemptedGossiped)
+            | (AttemptPending, AttemptPending)
+            | (Responded, Responded)
+            | (Failed, Failed) => Equal,
```

Comment:
Optional/Nitpick:

```suggestion
            _ if self == other => Equal,
            (Failed, Failed) => Equal,
```

Reply:
Applied in [cf7f618](https://github.com/ZcashFoundation/zebra/pull/6717/commits/cf7f618463462c8a30935e9e170574b8805c7d78)